src/node_url.cc

```diff
@@ -853,9 +863,18 @@ static url_host_type ParseHost(url_host* host,
   // First, we have to percent decode
   PercentDecode(input, length, &decoded);
 
-  // Then we have to punycode toASCII
-  if (!ToASCII(&decoded, &decoded))
-    goto end;
+  // Match browser behavior for ASCII only domains
+  // and do not run them through ToASCII algorithm.
+  if (IsAllASCII(&decoded)) {
+    // Lowercase aschii domains
+    for (size_t n = 0; n < decoded.size(); n++) {
```

Comment:
Technically it's not SIMD unless the compiler is somehow smart enough to automatically convert it to using SIMD. Anyway, I still think it would be worth benchmarking...